### PR TITLE
tests: Add new test helpers from timeline branch

### PR DESCRIPTION
**Pull request by artem-lunarg**
Some general helpers to minimize diff.

### CODE REVIEW

File: layers/sync/sync_stats.cpp [hunk 1]
```diff
@@ -26,8 +26,8 @@
 namespace syncval_stats {
 
 void Value32::Update(uint32_t new_value) { u32.store(new_value); }
-uint32_t Value32::Add(uint32_t n) { return u32.fetch_add(n); }
-uint32_t Value32::Sub(uint32_t n) { return u32.fetch_sub(n); }
+uint32_t Value32::Add(uint32_t n) { return u32.fetch_add(n) + 1; }
+uint32_t Value32::Sub(uint32_t n) { return u32.fetch_sub(n) - 1; }
```
Comment (jeremyg-lunarg) [inline layers/sync/sync_stats.cpp:30]: probably should have a comment about what the return values are for these, just to save people from having to go look up fetch_add & fetch_sub.

File: tests/framework/binding.h [hunk 2]
```diff
@@ -420,6 +420,7 @@ class Semaphore : public internal::NonDispHandle<VkSemaphore> {
     VkResult WaitKHR(uint64_t value, uint64_t timeout);
     VkResult Signal(uint64_t value);
     VkResult SignalKHR(uint64_t value);
+    uint64_t GetCounterValue(bool use_khr = false) const;
```
Comment (jeremyg-lunarg) [inline tests/framework/binding.h:423]: could `use_khr_` be part of the Semaphore object? Then you could get rid of the KHR versions of Wait and Signal as well.

Comment (artem-lunarg) [reply tests/framework/binding.h:423]: It's a good idea. I'd postpone this to a separate PR, currently we have other styles here like KHR in the signature name, and also submit helpers. Have a plan to update submit helpers too so can do as part of that task.

File: layers/sync/sync_stats.cpp (hunk 1 above)
Comment (artem-lunarg) [reply layers/sync/sync_stats.cpp]: done

File: tests/framework/binding.h (hunk 2 above)
Comment (jeremyg-lunarg) [reply tests/framework/binding.h:423]: Possibly we need an even more general solution. A similar question just popped up in internal chat. And I could use smarter dispatch in both CDL and VEL.

Comment (artem-lunarg) [reply tests/framework/binding.h:423]: @jeremyg-lunarg could you approve this nevertheless, I'm trying to move to the main branch what is not essential to timeline validation, for simplicity reasons.

**Review by jeremyg-lunarg — APPROVED** [review]

File: tests/framework/binding.h (hunk 2 above)
Comment (jeremyg-lunarg) [reply tests/framework/binding.h:423]: done

**Pull request merged**